maro/simulator/scenarios/vm_scheduling/frame_builder.py

```diff
@@ -3,21 +3,34 @@
 
 from maro.backends.frame import FrameBase, FrameNode
 
+from .cluster import Cluster
+from .data_center import DataCenter
 from .physical_machine import PhysicalMachine
+from .rack import Rack
+from .region import Region
+from .zone import Zone
 
 
-def build_frame(pm_amount: int, snapshots_num: int):
+def build_frame(snapshots_num: int, **kwargs):
     """Function to build vm_scheduling Frame.
 
     Args:
-        pm_amount (int): Number of physical machine.
         snapshot_num (int): Number of in-memory snapshots.
+        region_amount (int): Number of region.
+        zone_amount (int): Number of zone.
+        cluster_amount (int): Number of cluster.
+        pm_amount (int): Number of physical machine.
 
     Returns:
         VmSchedulingFrame: Frame instance for vm_scheduling scenario.
     """
     class VmSchedulingFrame(FrameBase):
-        pms = FrameNode(PhysicalMachine, pm_amount)
+        regions = FrameNode(Region, kwargs["region_amount"])
```

Comment:
https://stackoverflow.com/questions/1098549/proper-way-to-use-kwargs-in-python
https://stackoverflow.com/questions/12399803/how-to-check-if-a-key-in-kwargs-exists

It seems the key "region_amount" ... "pm_amount" are all needed?